cli/src/pcluster/utils.py

```diff
@@ -333,3 +336,24 @@ def get_attr(obj, attributes, default=None):
         if obj is None:
             return default
     return obj
+
+
+def yaml_load(s):
+    yaml.add_constructor(
+        tag=BaseResolver.DEFAULT_MAPPING_TAG, constructor=yaml_no_duplicates_constructor, Loader=SafeLoader
+    )
+    return yaml.safe_load(s)
+
+
+def yaml_no_duplicates_constructor(loader, node, deep=False):
+    """Check for duplicate keys."""
+
+    mapping = {}
+    for key_node, value_node in node.value:
```

Comment:
Duplicate keys are invalid, as per [YAML 1.2 specs](https://yaml.org/spec/1.2.2/), so this is something not related to pcluster schema but to YAML itself. 

Currently, we do not detect such duplicates out of the box because we rely on PyYAML, which is based on YAML 1.1 where duplicate avoidance is not enforced. 

Currently, we handle such syntax errors at the PyYAML level, not in pcluster schema, that0s why I put this logic here.

Reply:
I like the idea but unfortunately the path is not accessible when running the code from API. Let's keep the code as is. thanks!

Reply:
Clarified offline. We'll go with the proposal of surfacing the error message from the parser but keeping the logic at PyYAML level